my/coding/commits.py

```diff
@@ -1,38 +1,64 @@
 """
 Git commits data for repositories on your filesystem
 """
+REQUIRES = [
+    'gitpython',
+]
+
 
 import shutil
 import string
 from pathlib import Path
 from datetime import datetime, timezone
-from typing import List, NamedTuple, Optional, Dict, Any, Iterator, Set
+from dataclasses import dataclass, field
+import functools
+from typing import List, NamedTuple, Optional, Iterator, Set, Sequence
+
+
+from my.core import PathIsh, LazyLogger, make_config
+from my.core.cachew import cache_dir
+from my.core.common import mcachew
+from my.core.warnings import high
+
+
+from my.config import commits as user_config
+@dataclass
+class commits_cfg(user_config):
```

Comment:
name conflict with `def commits` below! But I guess doesn't hurt since it's basically internal to this module